### ISSUE TITLE
[HttpClient] port #29703 to .net core 2.1

### ISSUE THREAD

**Issue by daniel-white** [dotnet/corefx#40451]
Can we please have #29703 back to 2.1? Right now we have a customer that is using a Salesforce sandbox that has a 2 letter start before 2 hypens as the subdomain (`xx--<sandbox>.<instance>.salseforce.com>`). We tried turning off the sockets HTTP handler, but this has regressed our performance switching to the CURL implementation.

thx

**Comment by stephentoub** [comment]
cc: @karelz, @davidsh

**Comment by daniel-white** [comment]
We just rolled back to using the new sockets handler and it improved our throughput as expected. I would contend that this issue is somewhat of a regression from the old behavior which should be a reasonable justification for backporting

**Comment by karelz** [comment]
I'd be fine running it by servicing folks for bar check.
@eiriktsarpalis is it something you could help with please? @davidsh can provide guidance. Thanks!

**Comment by karelz** [comment]
Hmmm, looking at the original bug, it seems it was general Globalization bug that would need to be ported. @tarekgh any thoughts?

**Comment by tarekgh** [comment]
> Hmmm, looking at the original bug, it seems it was general Globalization bug that would need to be ported. @tarekgh any thoughts?

We just tweaked a special case in IDN to serve http/Uri failing cases. The fix is simple but we are turning some failing case (throwing exception) into succeeding case which is kind of a breaking changing. I wouldn't expect any real user can depend on the throwing exception or even if we treat such cases as a real breaking change.

**Comment by daniel-white** [comment]
@tarekgh I’d be fine with putting this behind a flag for compatibility

**Comment by tarekgh** [comment]
>  I’d be fine with putting this behind a flag for compatibility

I don't think we do that in servicing releases. Also, that will not be just porting but also adding more stuff that will need more testing and validation. I mean will add more risk to the fix.

To be clear I am not pushing back on the porting. I am just explaining the risks. @karelz please advise.

**Comment by karelz** [comment]
Thanks @tarekgh. I'd still run it by servicing shiproom for bar check first. We will know more afterwards.

Coming at flags is not something we do in .NET Core.

**Comment by tarekgh** [comment]
@daniel-white we have consulted the servicing shiproom. Because 2.1 is LTS release and any change there has to hold to a high bar, requesting the change there need to be done through Microsoft Support for evaluating the broken customer case and risks. If you think this is critical to the any specific customer, please go ahead and file a case to Microsoft Support.

**Comment by daniel-white** [comment]
@tarekgh thanks! i completely understand. i was unaware that was the course to take. i have opened it on that side.
